pkgs/by-name/au/autobrr/package.nix

```diff
@@ -0,0 +1,71 @@
+{
+  lib,
+  cacert,
+  buildGoModule,
+  fetchFromGitHub,
+  stdenvNoCC,
+  nix-update-script,
+  nodePackages
+}:
+
+let
+  pname = "autobrr";
+  version = "1.36.0";
+  src = fetchFromGitHub {
+    owner = "autobrr";
+    repo = "autobrr";
+    rev = "v${version}";
+    hash = "sha256-HQuiREj84nTv9EpKaFKNmCLLmHRko3QR4qoa6ICWwHY=";
+  };
+
+  autobrr-web = stdenvNoCC.mkDerivation {
+    pname = "${pname}-web";
+    inherit src version;
+
+    nativeBuildInputs = [
+      nodePackages.pnpm
+      cacert
+    ];
+
+    buildPhase = ''
+      runHook preBuild
+
+      export HOME=$(mktemp -d)
+      pnpm --dir web install
+      pnpm --dir web run build
+
+      mv web/dist $out
+
+      runHook postBuild
+    '';
+
+    dontInstall = true;
+
+    outputHashMode = "recursive";
+    outputHash = "sha256-IOrW26Nq+9PYWlzUSpPBfkv3jzs5VlfF0JVFaUCDMmw=";
+  };
```

Comment:
So at least we're not using an FOD to store the `pnpm` dependencies. But I still worry that the output is not bit-for-bit reproducible.

My recommendation is to:
1. Try to use `pnpm-export` and use `buildNpmPackage` or `mkYarnPackage`.
2. If `pnpm-export` doesn't work, just generate a lock file manually.

In either case, this would mean vendoring the lock file to the repository.

Reply:
I believe this derivation should be bit-for-bit reproducible. Different versions of pnpm might have different store or cache formats, but each module in `node_modules` will be consistent with its hash in `pnpm-lock.yaml`. `dist` just contains the final bundle created by Vite. Why would that change because the dependencies are grabbed with a different package manager?

Reply:
What makes you believe the output of `pnpm --dir web run build` is reproducible? Do TypeScript and Vite claim to be reproducible?

Reply:
```
error: build of '/nix/store/cfcl0gpb9p5nnphw25g9mc6hifjpf3wl-autobrr-web-1.36.0.drv' on 'ssh://nix@willard' failed: hash mismatch in fixed-output derivation '/nix/store/cfcl0gpb9p5nnphw25g9mc6hifjpf3wl-autobrr-web-1.36.0.drv':
         specified: sha256-IOrW26Nq+9PYWlzUSpPBfkv3jzs5VlfF0JVFaUCDMmw=
            got:    sha256-SkwSKFEZAmjVnaSowIbrdH667vB5WqNrPuRs/Yh6BLc=
```

Reply:
`pnpm --dir web run build` just runs `tsc && vite build`, as specified in `package.json`. Neither of these claim to be reproducible build tools (I don't think), but I believe that `buildNpmPackage` or `mkYarnPackage` would execute those commands in the same way. Please correct me if I'm wrong.

Reply:
May I ask if you did anything to cause the hash mismatch? Or did you just try to build it on your machine

Reply:
The latter.



> Neither of these claim to be reproducible build tools (I don't think), but I believe that `buildNpmPackage` or `mkYarnPackage` would execute those commands in the same way.

Yes but those aren't used in FODs.